client/src/hooks/use-modal.ts

```
@@ -0,0 +1,296 @@
+    onOpen,
+    onClose,
+    animationDuration = 300
+  } = config;
+
+  // Validate configuration
+  if (persistent && !storageKey) {
+    console.warn('useModal: storageKey is required when persistent is true');
+  }
+
+  // Initialize state
+  const [state, setState] = useState<ModalState>(() => {
```

Biome (2.1.2)

[error] This hook is being called from a nested function, but all hooks must be called unconditionally from the top-level component.

This is the call path until the hook.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)